src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenshotCapture.cs

```diff
@@ -290,82 +290,102 @@ private IEnumerator UpdateGPUData(int frame, Action<(byte[], int, int)?> onCompl
                     }
                 }
             }
-
-            // wait for end of frame or graphics data will be incorrect
-            yield return new WaitForEndOfFrame();
-
-            // get this in there so we don't get duplicate requests out for the same frame
-            if (GPUReadbackRequests.TryAdd(frame, null))
+            
+            // If we are running in -nographics mode, the async task fails, causing an exception inside
+            // the AsyncGPUReadback.Request that is difficult to catch. This ensures that the screenshot
+            // is only taken when we have graphics
+            if (SystemInfo.graphicsDeviceType != GraphicsDeviceType.Null)
```

Comment:
(Just thinking aloud to myself here) On lines 284-285, I was worried that these actions wouldn't be completed properly if I placed the check here, but it seems like that's not the case... and should be fine

Reply:
Turns out I was correct in my worry and incorrect in whether this would be fine XD TLDR; because of how I wrote the original code, the onComplete handlers never get called, and so while screenshots are not taken, the handlers that would have gotten called which handle writing the ticks also never get called. The main change is now in the `GetCurrentScreenshotWithCallback`, which will just immediately invoke the completion handler, fixing the bug I was seeing. I've left this check here though as a safety measure since I wanted to keep this check near the code that actually does the readback request too.

Reply:
Please remove this code block.  It is un-necessary code with your other check in the correct place and actually is confusing.  IF you insist on leaving this code, then it should move to the start of this function before the synclock and would be similar to the code in the calling function where it checks and instantly invokes the completion action.  It would be cleaner to just remove this though